src/www/views/servers_view/server_connection_indicator/index.ts

```diff
@@ -11,284 +11,191 @@
-          height: 48px;
-          width: 48px;
-        }
-        .expanded #medium,
-        .expanded #medium-grey {
-          top: 36px;
-          left: 36px;
-          height: 120px;
-          width: 120px;
+    `,
+    ...CIRCLE_SIZES.map(
+      /* prettier-ignore */
+      circleSize => css`
+        .circle-${circleSize} {
+          transform: var(--circle-${circleSize}-scale);
+          animation-delay: var(--circle-${circleSize}-delay);
         }
-        .expanded #large,
-        .expanded #large-grey,
-        .expanded #large-zero {
-          top: 0;
-          left: 0;
-          height: 192px;
-          width: 192px;
-        }
-      }
-      /* rtl:end:ignore */
-    </style>
-    <div id="container" class$="[[expandedClassName]]">
-      <img id="small-grey" src$="[[rootPath]]assets/disc_grey.png" class$="grey {{animationState}}" />
-      <img id="small" src$="[[rootPath]]assets/disc_color.png" class$="green {{animationState}}" />
-      <img id="medium-grey" src$="[[rootPath]]assets/disc_grey.png" class$="grey {{animationState}}" />
-      <img id="medium" src$="[[rootPath]]assets/disc_color.png" class$="green {{animationState}}" />
-      <img id="large-grey" src$="[[rootPath]]assets/disc_grey.png" class$="grey {{animationState}}" />
-      <img id="large-zero" src$="[[rootPath]]assets/disc_empty.png" class$="{{state}}" />
-      <img id="large" src$="[[rootPath]]assets/disc_color.png" class$="green {{animationState}}" />
-    </div>
-  `;
-
-  animationStartMS: number;
-
-  @property({type: String}) rootPath: string;
-  @property({type: Boolean}) expanded: boolean;
-
-  @property({type: String, observer: "syncAnimationState"}) state: ServerConnectionState;
-  @property({type: String}) animationState: ServerConnectionState = ServerConnectionState.INITIAL;
-
-  @computed("expanded")
-  get expandedClassName() {
-    return this.expanded ? "expanded" : "";
-  }
 
-  @computed("state")
-  get shouldAnimate() {
-    return this.isAnimationState(this.state);
-  }
+        .circle-${circleSize}.circle-connecting,
+        .circle-${circleSize}.circle-reconnecting,
+        .circle-${circleSize}.circle-disconnecting {
+          animation-name: circle-${circleSize}-rotate-with-pause;
+        }
 
-  @computed("animationState")
-  get isAnimating() {
-    return this.isAnimationState(this.animationState);
-  }
+        /* Do not mirror animation for RTL languages */
+        /* rtl:begin:ignore */
+        @keyframes circle-${circleSize}-rotate-with-pause {
+          0% {
+            transform: rotate(0deg) var(--circle-${circleSize}-scale);
+          }
+          60%,
+          100% {
+            transform: rotate(360deg) var(--circle-${circleSize}-scale);
+          }
+        }
+        /* rtl:end:ignore */
+      `
+    ),
+  ];
 
-  // @ts-ignore
-  private syncAnimationState() {
+  render() {
     if (this.shouldAnimate) {
-      return this.startAnimation();
+      this.startAnimation();
```

Comment:
By calling this on render, not on connection state change, you are modifying the logic. Is the animation restarting on every render? It's hard to figure out whether it's doing the right thing. Could you change it back to when the property is modified?

Reply:
Hm, I'm going off the ethos of "Lit is Polymer inspired by React" here - which, this is a very common pattern to see in React, for better or worse (I hate that state resolution gets mixed in with the render path but this is what everyone has agreed is best). For what it's worth I've tested it and it works.

I should learn more about Lit - Lit doesn't have observables anymore. I'll try to figure out the proper lifecycle callback to use (seems like `willUpdate`).

Reply:
Perhaps a simple solution is to create a connectionState setter that updates the animation state?
willUpdate should work too

Reply:
Yeah just pushed the willUpdate change